[TypeScript - Angular] Syntax correction in the configuration.ts template

nnworkspace opened the pull request:
The generated `configuration.ts` contains error and warnings from my TypeScript 4.1.5 compiler. things like `this.apiKeys['api_key']` is not allowed and I changed it to `this.apiKeys?.api_key`, and edited the Mustache template accordingly here. 

Note: When the property name contains "-" sign, the dot syntax won't work. I had problem converting `this.apiKeys['X-API-Key']` to `this.apiKeys.X-API-Key`. That's why I used `ts-ignore` once in the file.

<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

nnworkspace commented:
The error from circleci looks like this:
```
[SEVERE] built_value_generator:built_value on lib/model/enum_arrays.dart:
An error `FormatterException` occurred while formatting the generated source for
  `package:openapi/model/enum_arrays.dart`
which was output to
  `lib/model/enum_arrays.built_value.g.part`.
This may indicate an issue in the generator, the input source code, or in the
source formatter.
Could not format because the source could not be parsed:

line 43, column 5: Expected an identifier.
   ╷
43 │ '\\$': 'dollar',
   │     ^
   ╵
line 39, column 15: Expected an identifier.
   ╷
39 │ 'dollar': '\\$',
   │               ^
   ╵
[INFO] 13.9s elapsed, 47/63 actions completed.
[INFO] 15.0s elapsed, 89/104 actions completed.
[INFO] Running build completed, took 16.0s

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 110ms

[SEVERE] Failed after 16.1s
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (pub-run-build-runner) on project DartDioPetstoreClientLibFakeTests: Command execution failed. Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :DartDioPetstoreClientLibFakeTests
```
But this has really nothing to do with my patch. I only edited one Mustache template in the Typescript-Angular part. I did not touch Dart at all....Why is this failing?

morphlne commented:
Faced the same problem - just merge last changes from master and this issue will gone

wing328 commented:
Yes, please merge the latest master into your branch by running
```
git pull https://github.com/openapitools/openapi-generator
```

wing328 commented:
> The generated configuration.ts contains error and warnings from my TypeScript 4.1.5 compiler. things like this.apiKeys['api_key'] is not allowed and I changed it to this.apiKeys?.api_key, and edited the Mustache template accordingly here.

Does the new syntax work with older versions of TS (e.g 3.x) ?

wing328 commented:
> Note: When the property name contains "-" sign, the dot syntax won't work. I had problem converting this.apiKeys['X-API-Key'] to this.apiKeys.X-API-Key. That's why I used ts-ignore once in the file.

Besides using `.` to access the value of the hash (inteface) `apiKeys`, is there any other way? "X-API-KEY" is perfectly valid should be supported.

nnworkspace commented:
> Yes, please merge the latest master into your branch by running
> 
> ```
> git pull https://github.com/openapitools/openapi-generator
> ```

Yes, just did what you said and the CI checks are running now.

nnworkspace commented:
> > Note: When the property name contains "-" sign, the dot syntax won't work. I had problem converting this.apiKeys['X-API-Key'] to this.apiKeys.X-API-Key. That's why I used ts-ignore once in the file.
> 
> Besides using `.` to access the value of the hash (inteface) `apiKeys`, is there any other way? "X-API-KEY" is perfectly valid should be supported.

Yes, X-API-KEY is supported in the template I modified. It still uses old syntax, not dot. But only at this place I used a tag to ask TS compiler to please ignore this. All the other places I have already changed to the . syntax. Hope it works!

wing328 commented:
Can you please resolve the merge conflicts when you've time? I'll then tag TypeScript technical committee to review the change.
